File: SharpDevelop/VBasic/w7cpframe/clsPfMemberLoad.vb
'
' Copyright (c)2016 S C Harrison
' Refer to License.txt for terms and conditions of use.
'
Option Explicit On

Public Class clsPfMemberLoad



#Const isUsingExcel = 0

    Public key As Integer

    Public mem_no As Byte
    Public lcode As Byte
    Public f_action As Byte
    Public ld_mag1 As Double     '.. member load magnitude 1 ..
    Public ld_mag2 As Double     '.. member load magnitude 2 ..
    Public start As Double       '.. dist from end_1 to start/centroid of load ..
    Public cover As Double      '.. dist that a load covers ..

    Sub initialise()
        mem_no = 0
        lcode = 0
        f_action = 0
        ld_mag1 = 0
        ld_mag2 = 0
        start = 0
        cover = 0
    End Sub

    Sub setValues(ByVal LoadKey As Integer, ByVal memberKey As Integer, ByVal LoadType As Integer, ByVal ActionKey As Integer _
                             , ByVal LoadMag1 As Double, ByVal LoadMag2 As Double, ByVal LoadStart As Double, ByVal LoadCover As Double)

        key = LoadKey
        mem_no = memberKey
        lcode = LoadType
        f_action = ActionKey
        ld_mag1 = LoadMag1
        ld_mag2 = LoadMag2 'xla version only
        start = LoadStart
        cover = LoadCover
    End Sub

    Function sprint() As String
        sprint = StrLPad(Format(key, "##0"), 8) _
                   & StrLPad(Format(mem_no, "##0"), 6) _
                   & StrLPad(Format(lcode, "##0"), 6) _
                   & StrLPad(Format(f_action, "##0"), 6) _
                   & StrLPad(Format(ld_mag1, "0.000"), 15) _
                   & StrLPad(Format(start, "0.000"), 15) _
                   & StrLPad(Format(cover, "0.000"), 12)

    End Function

    Sub cprint()
        Console.WriteLine(sprint)
    End Sub

    Sub fprint(ByVal fp As Integer)
        Print(fp, sprint)
    End Sub


#If isUsingExcel = 1 Then

    Sub wbkPrint(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            .Offset(RecNum, 0).Value = key
            .Offset(RecNum, 1).Value = mem_no
            .Offset(RecNum, 2).Value = lcode
            .Offset(RecNum, 3).Value = f_action
            .Offset(RecNum, 4).Value = ld_mag1
            .Offset(RecNum, 5).Value = start
            .Offset(RecNum, 6).Value = cover
        End With
    End Sub

    Sub wbkRead(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            key = .Offset(RecNum, 0).Value
            mem_no = .Offset(RecNum, 1).Value
            lcode = .Offset(RecNum, 2).Value
            f_action = .Offset(RecNum, 3).Value
            ld_mag1 = .Offset(RecNum, 4).Value
            ld_mag2 = .Offset(RecNum, 5).Value
            start = .Offset(RecNum, 6).Value
            cover = .Offset(RecNum, 7).Value
        End With
    End Sub

#End If

End Class

File: Eclipse/w7cpframe/src/clsPfMemberLoad.java
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
import java.io.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;


    class clsPfMemberLoad
    {


        public int key = 0;

        public int mem_no = 0;
        public int lcode = 0;
        public int f_action = 0;
        public double ld_mag1 = 0;     //.. member load magnitude 1 ..
        public double ld_mag2 = 0;     //.. member load magnitude 2 ..
        public double start = 0;       //.. dist from end_1 to start/centroid of load ..
        public double cover = 0;       //.. dist that a load covers ..



        public void initialise()
        {
            key = 0;
            mem_no = 0;
            lcode = 0;
            f_action = 0;
            ld_mag1 = 0;
            ld_mag2 = 0;
            start = 0;
            cover = 0;
        }

        public void setValues(int LoadKey, int memberKey, int LoadType, int ActionKey, double LoadMag1, double LoadStart, double LoadCover)
        {
            key = LoadKey;
            mem_no = memberKey;
            lcode = LoadType;
            f_action = ActionKey;
            ld_mag1 = LoadMag1;
            //ld_mag2 = LoadMag2; //xla version only
            start = LoadStart;
            cover = LoadCover;
        }

        public String sprint()
        {
            String s;

            s = "";
            s = s + String.format("%8d",key);
            s = s + String.format("%6d",mem_no);
            s = s + String.format("%6d",lcode);
            s = s + String.format("%6d",f_action);
            s = s + String.format("%15.4f",ld_mag1);
            s = s + String.format("%15.4f",start);
            s = s + String.format("%12.3f",cover);

            return s;

        }

        public void cprint()
        {
            System.out.println(sprint());
        }

        public void fprint(BufferedWriter fp)
        {
            try {
				fp.write(sprint());
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
        }


        public void fgetData(BufferedReader fp)
        {
            String s="";

            System.out.println("fgetData ...");

            try {
				s = fp.readLine();
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
            System.out.println(s);
            sgetData(s);

            System.out.println("... fgetData");
        }


        public void sgetData(String s)
        {
            String[] dataflds = new String[10]; //(0 To 9);
            int i, n;


            System.out.println("sgetData ...");

            //WScript.Echo( s);

            //Trim Trailing Spaces from input String
            String regExpTrimPattern = "^\\s+|\\s+$"; //trim trailing spaces          
            Pattern regExpPatternObj = Pattern.compile(regExpTrimPattern);
            Matcher regExpMatches = regExpPatternObj.matcher(s);
            String regResult = regExpMatches.replaceAll("");
            
            //Get Data fields from input String
            String regExpNumberPattern = "-?\\d+(?:[,.]\\d+)?";
            regExpPatternObj = Pattern.compile(regExpNumberPattern);
            regExpMatches = regExpPatternObj.matcher(regResult);
            i=0;
            while (regExpMatches.find()){
            	dataflds[i] = regExpMatches.group().trim();
            	i = i+1;
            }


            n = dataflds.length;
            for (i = 0; i < n; i++)
            {
                System.out.println(i + "<" + dataflds[i] + ">");
            }

            key = Integer.parseInt(dataflds[0]);
            mem_no = Integer.parseInt(dataflds[1]);
            lcode = Integer.parseInt(dataflds[2]);
            f_action = Integer.parseInt(dataflds[3]);
            ld_mag1 = Double.parseDouble(dataflds[4]);

            ld_mag2 = ld_mag1;

            start = Double.parseDouble(dataflds[5]);
            cover = Double.parseDouble(dataflds[6]);

            System.out.println("... sgetData");
        }

    } //class
